### PR TITLE
pre-commit: Add more checks

### PR DESCRIPTION
**Pull request by armenzg**
A lot of automatic fixes. I tried to enabled one check per commit (not initially).

The only changes you should really look at are the one on the pre-commit config file.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -2,22 +2,22 @@ repos:
...
     hooks:
-      # - id: trailing-whitespace
-      # - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
       - id: check-yaml
       - id: debug-statements
-      # - id: double-quote-string-fixer
       # - id: name-tests-test
+      #   args: ["--django"]
```
Comment (armenzg) [inline .pre-commit-config.yaml:10]: I'm leaving this for the future. For when I actually want to run test and want to make sure things are actually running.

File: backend-py/src/api/endpoints/base.py [hunk 2]
```diff
@@ -16,15 +16,17 @@ def register_api(
         url,
         defaults={pk: None},
         view_func=view_func,
-        methods=["GET", ]
+        methods=[
+            "GET",
+        ],
```
Comment (asottile-sentry) [inline backend-py/src/api/endpoints/base.py:21]: I know black is supposed to be "don't talk about formatting" -- but if you remove the comma here black will produce better formatting (same for the one below)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (asottile-sentry) [inline .pre-commit-config.yaml:10]: if you're using pytest to run these I would not recommend the `--django` setting -- it allows files to sneak through without being run

**Review by asottile-sentry — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (armenzg) [reply .pre-commit-config.yaml:10]: I left it commented because I need to investigate what the complains are. I may need to rename things around.

Comment (asottile-sentry) [reply .pre-commit-config.yaml:10]: the default or `--pytest-test-first` are what I'd recommend :+1:

**Pull request merged**